src/main.py

```diff
@@ -1,7 +1,109 @@
-from remote_connector_dao.plot_data_test import test_run
-from remote_connector_dao.get_signal_tests import test_signal_run
-from remote_connector_dao.test_real_time_graph import run_real_time_graph
+import sys
+from PyQt5 import QtWidgets
+from AppController import AppController, CloseConnectionPrams
+from shared.constants.constants import TRIGGER_VOLTAGE
+from time_tagger.connection.repository import ConnectionRepository
+from time_tagger.connection.service import (
+    TimeTaggerAddressInfo,
+    ConnectionService,
+)
+from time_tagger.hardware_properties.dao import (
+    SetTriggerLevelParams,
+    TimeTaggerHardwarePropertiesDao,
+)
+from time_tagger.measurement.repository import MeasurementRepository
+from time_tagger.measurement.service import MeasurementService
+from time_tagger.builder import TimeTaggerBuilder
+from time_tagger.connection.dao import ConnectionDao
+from ui.MainWindow import MainWindow
 
-# test_run()
-run_real_time_graph()
-# test_signal_run()
+"""
+    This file is a mess because I am only using this for testing.
+"""
+
+channels = [1, 2]
+connection_data = ConnectionRepository()
+
+time_tagger_hardware_properties_service = TimeTaggerHardwarePropertiesDao()
+
+connection_dao = ConnectionDao()
+
+time_tagger_network_connection_service = ConnectionService(
+    connection_data, time_tagger_hardware_properties_service, connection_dao
+)
+
+time_tagger_builder = TimeTaggerBuilder()
+
+measurement_data = MeasurementRepository()
+
+
+time_tagger_measurement_service = MeasurementService(measurement_data)
+
+app_controller = AppController(
+    time_tagger_network_connection_service=time_tagger_network_connection_service,
+    time_tagger_measurement_service=time_tagger_measurement_service,
+    time_tagger_hardware_properties=time_tagger_hardware_properties_service,
+)
+
+# The first connection should fails because there is no server on this port.
+req = app_controller.connect_to_time_taggers_network(
+    [
+        TimeTaggerAddressInfo(
+            host_address="192.168.10.100",
+            port="41101",
+            time_tagger_name="Alice_tagger",
+        ),
+        TimeTaggerAddressInfo(
+            host_address="192.168.10.100",
+            port="43101",
+            time_tagger_name="Bob_tagger",
+        ),
+    ]
+)
+
+print(req, end="\n\n")
+
+print(req.connection_failed_devices, "connection failed devices", end="\n\n")
+
+tagger = req.connected_devices[0].tagger_proxy
+connected_device_name = req.connected_devices[0].time_tagger_name
```

Comment:
Sent you some videos proposing the creation of an entity to help encapsulating the info about a device:

```
@dataclass
class Tagger: # feel free to change the name to something better
    proxy: 
    name: str
    device_serial: str

    def get_count_rates(self, channels: list[int]):
        ...


    # we can then merge the class TimeTaggerHardwarePropertiesDao here
    # no need to pass the requests through the Controller
    def set_trigger_level(self, channels_voltage: dict[int, float]):
         # we can then store the channels in this object, so we don't need to keep passing it along the tagger in every method invocation 
         ...

taggers : list[Tagger] = app_controller.connect_to_time_taggers_network(
    [
        TimeTaggerAddressInfo(
            host_address="192.168.10.100",
            port="41101",
            time_tagger_name="Alice_tagger",
        ),
        TimeTaggerAddressInfo(
            host_address="192.168.10.100",
            port="43101",
            time_tagger_name="Bob_tagger",
        ),
    ]
)
`
``